File: services/ts-express-superstruct/v2/routing.ts
import {Router} from 'express'
import {Request, Response} from 'express'
import * as t from './../superstruct'
import * as models from './models'
import {EchoService} from './echo_service'

export let echoRouter = (service: EchoService) => {
    let router = Router()

    router.post('/echo/body', async (request: Request, response: Response) => {
        var body: models.Message
        try {
            body = t.decode(models.TMessage, request.body)
        } catch (error) {
            response.status(400).send()
            return
        }
        try {
            let result = await service.echoBody({body})
            switch (result.status) {
                case 'ok':
                    response.status(200).type('json').send(JSON.stringify(t.encode(models.TMessage, result.data)))
            }
        } catch (error) {
            response.status(500).send()
        }
    })

    return router
}

File: services/go/v2/routing.go
package v2

import (
	"encoding/json"
	"github.com/husobee/vestigo"
	log "github.com/sirupsen/logrus"
	"net/http"
	"test-service/v2/echo"
	"test-service/v2/models"
)

func AddEchoRoutes(router *vestigo.Router, echoService echo.Service) {
	logEchoBody := log.Fields{"operationId": "echo.echo_body", "method": "POST", "url": "/v2/echo/body"}
	router.Post("/v2/echo/body", func(res http.ResponseWriter, req *http.Request) {
		log.WithFields(logEchoBody).Info("Received request")
		var body models.Message
		err := json.NewDecoder(req.Body).Decode(&body)
		if err != nil {
			log.WithFields(logEchoBody).Warnf("Decoding body JSON failed: %s", err.Error())
			res.WriteHeader(400)
			log.WithFields(logEchoBody).WithField("status", 400).Info("Completed request")
			return
		}
		response, err := echoService.EchoBody(&body)
		if err != nil {
			log.WithFields(logEchoBody).Errorf("Error returned from service implementation: %s", err.Error())
			res.WriteHeader(500)
			log.WithFields(logEchoBody).WithField("status", 500).Info("Completed request")
			return
		}
		if response == nil {
			log.WithFields(logEchoBody).Errorf("No result returned from service implementation")
			res.WriteHeader(500)
			log.WithFields(logEchoBody).WithField("status", 500).Info("Completed request")
			return
		}
		res.WriteHeader(200)
		json.NewEncoder(res).Encode(response)
		log.WithFields(logEchoBody).WithField("status", 200).Info("Completed request")
		return
	})
}
